agnpy/emission_regions/blob.py

```diff
@@ -393,3 +397,50 @@ def u_ph_synch(self):
             * np.trapz(np.power(self.gamma_e, 2) * self.n_e(self.gamma_e), self.gamma_e)
         )
         return u_ph.to("erg cm-3")
+
+    def n_e_time_evolution(self, energy_loss_function, time, subintervals_count=10):
+        """Performs the time evolution of the electron distribution inside the blob.
+
+        Parameters
+        ----------
+        energy_loss_function : function
+            the function to be used for calculation of energy loss rate per gamma values
+        time : `~astropy.units.Quantity`
+            total time for the calculation
+        subintervals_count : int
+            optional number defining how many equal-length subintervals the total time will be split into
+        """
+        if (subintervals_count <= 0):
+            raise ValueError("subintervals_count must be > 0")
+
+        unit_time_interval = time / subintervals_count
+
+        def gamma_recalculated_after_loss(gamma):
+            old_energy = (gamma * mec2).to("erg")
+            energy_loss_per_time = energy_loss_function(gamma).to("erg s-1")
+            energy_loss = (energy_loss_per_time * unit_time_interval).to("erg")
+            new_energy = old_energy - energy_loss
+            if np.any(new_energy < 0):
```

Comment:
this we discussed in person. If the new energy is negative then the code has failed miserably with way too large time step. However if the loss is e.g. 50% is also is very large, because energy losses at E/2 will be different. And limiting the global step size can be inefficient because at low energies nothing is happening. 
The most efficient solution might be try a given step, but at energies where it is too large (e.g. change of energy is more then 10%) the step is divided iteratively by 2 until the change is small enough at this energy.

Reply:
I'll work on it in the separate branch.